### PR TITLE
Adapt the tests for the new rust agent configuration format

### PR DESCRIPTION
**Pull request by ansasaki**
Modify the tests to be compatible with the TOML format used in the rust agent. The main differences are:

- Booleans: on TOML format they are case sensitive. Accepted values are `true` and `false`
- Strings: on TOML format they need to be double quoted.

These changes should be compatible with INI format.

This branch is temporarily used for testing the PR:
https://github.com/keylime/rust-keylime/pull/449

### CODE REVIEW

**Comment by ansasaki** [comment]
This made me find 2 details on python keylime agent:

- For the `ek_handle` option, it parses the handle as a number instead of a string. So, we cannot use quoted string like `ek_handle = "0x81010001"`
- For the `run_as` option, it does not strip the quotes from the string, making `run_as = "keylime:tss"` invalid. We need to use unquoted strings.

File: functional/basic-attestation-with-unpriviledged-agent/test.sh [hunk 1]
```diff
@@ -50,16 +45,20 @@ rlJournalStart
         rlRun "limeStartRegistrar"
         rlRun "limeWaitForRegistrar"
         # do special configuration for the agent
-        rlRun "limeUpdateConf ${AGENT_CONFIG_SECTION} run_as ${AGENT_USER}:${AGENT_GROUP}"
+        if limeIsPythonAgent; then
+            rlRun "limeUpdateConf agent run_as ${AGENT_USER}:${AGENT_GROUP}"
+        else
+            rlRun "limeUpdateConf agent run_as '\"${AGENT_USER}:${AGENT_GROUP}\"'"
```
Comment (kkaarreell) [inline functional/basic-attestation-with-unpriviledged-agent/test.sh:51]: Is this something that won't work with Python agent?

**Comment by kkaarreell** [comment]
LGTM.. Though later we may need to update other tests not run in CI

File: functional/basic-attestation-with-unpriviledged-agent/test.sh (hunk 1 above)
Comment (ansasaki) [reply functional/basic-attestation-with-unpriviledged-agent/test.sh:51]: After trying to make them the same, I realized that the python agent does not strip the quotes from the strings obtained from the configuration in all cases (this being one of them)

So no, while the TOML format requires the quotes, the python agent does not work correctly when the strings are quoted.

Comment (kkaarreell) [reply functional/basic-attestation-with-unpriviledged-agent/test.sh:51]: I see.

**Comment by ansasaki** [comment]
I'll make more changes as Sergio requested the configuration files and snippets to be installed following the python components structures for consistency. So the rust agent configuration will be placed at `/etc/keylime/agent.conf` and the snippets at `/etc/keylime/agent.conf.d`

**Comment by kkaarreell** [comment]
Since you are already updating the Rust agent install task, could you incorporate the configuration using environment variables as we do for Python?
https://github.com/RedHat-SP-Security/keylime-tests/blob/a5ad281392b1cea5d90f26c225e9996f2b9f102c/setup/install_upstream_keylime/test.sh#L7

This way, next time one could override repo on a test plan side, e.g. in a Rust agent PR only.

**Comment by ansasaki** [comment]
> Since you are already updating the Rust agent install task, could you incorporate the configuration using environment variables as we do for Python?
> 
> https://github.com/RedHat-SP-Security/keylime-tests/blob/a5ad281392b1cea5d90f26c225e9996f2b9f102c/setup/install_upstream_keylime/test.sh#L7
> 
> This way, next time one could override repo on a test plan side, e.g. in a Rust agent PR only.

Sure, I'll include this change

**Pull request merged**